src/schema/duration.rs

```diff
@@ -0,0 +1,358 @@
+use std::io::{Read, Write};
+use yaserde::{YaDeserialize, YaSerialize};
+
+#[derive(Default, PartialEq, Debug)]
+pub struct Duration {
+    pub is_negative: bool,
```

Comment:
Try and be consistent with all `bool` naming.  Either they all use `is_` or they don't.